src/emailcontactstorage.cpp

```diff
@@ -0,0 +1,144 @@
+#include <algorithm>
+#include <QMailStore>
+#include "emailcontactstorage.h"
+
+Q_GLOBAL_STATIC(EmailContactStorage, storage)
+EmailContactStorage* EmailContactStorage::instance()
+{
+    if (storage != nullptr && storage->m_isInit == false)
+        storage->initialize();
+    return storage;
+}
+
+
+void EmailContactStorage::initialize()
```

Comment:
Nothing updates the contacts after start?

Reply:
The add() function adds contacts.

For one session, only list adding.
Deleting a contact should not remove it from the list (there may be an erroneous deletion)
Renaming should keep the previous contact and add a new one.

Reply:
I mean you start the app, write emails to 5 people, start writing a new one -> should include the suggestions from the earlier 5, right?

Reply:
Right.
the names and emails of these five should be added.